pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -826,7 +826,29 @@ func getCRICadvisorStats(infos map[string]cadvisorapiv2.ContainerInfo) map[strin
 		if !isPodManagedContainer(&info) {
 			continue
 		}
-		stats[path.Base(key)] = info
+		stats[extractIDFromCgroupPath(key)] = info
 	}
 	return stats
 }
+
+func extractIDFromCgroupPath(cgroupPath string) string {
+	//case0: /kubepods/burstable/pod2fc932ce-fdcc-454b-97bd-aadfdeb4c340/9be25294016e2dc0340dd605ce1f57b492039b267a6a618a7ad2a7a58a740f32
+	id := path.Base(cgroupPath)
+
+	//case1: "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod2fc932ce_fdcc_454b_97bd_aadfdeb4c340.slice/cri-containerd-aaefb9d8feed2d453b543f6d928cede7a4dbefa6a0ae7c9b990dd234c56e93b9.scope"
```

Comment:
I don't think this format is supported (it looks like mix between systemd and cgroupfs name) as you mentioned in https://github.com/kubernetes/kubernetes/issues/103366#issuecomment-889286696

cgroup driver between kubelet and container runtime should match, so I don't think this is a valid case that should be supported.

Reply:
Hi @bobbypage , actually I am not sure whether this format is due to mix use of systemd and cgroupfs since I cannot reproduce this case now.

For the pod part `/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod2fc932ce_fdcc_454b_97bd_aadfdeb4c340.slice`, it looks like a systemd driver. 

But for the container part `cri-containerd-aaefb9d8feed2d453b543f6d928cede7a4dbefa6a0ae7c9b990dd234c56e93b9.scope`, I am not sure how it is generated, because in the cgroupfs case, we didn't see `cri-containerd-` prefix or  `.scope` suffix

Reply:
Hi @bobbypage, I prefer to keep that case since we're not sure whether it's not a invalid case. And as suggested by @haircommander , I changed the code not to limit to suffix `.scope` but use `.` to split the key.